core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -532,7 +532,8 @@ public PersistenceState getPersistenceState(long fileId) throws FileDoesNotExist
       Inode<?> inode = null;
       if (inodePath.fullPathExists()) {
         inode = inodePath.getInode();
-        if (inode.isDirectory() && ((InodeDirectory) inode).isDirectChildrenLoaded()) {
+        if ((inode.isDirectory() && ((InodeDirectory) inode).isDirectChildrenLoaded()) || inode
+            .isFile()) {
```

Comment:
Without this, we  will introduce a ufs.exists() call for each file that is in Alluxio.


@jsimsa @gpang  I *think* 1.1 changed the behavior of getFileInfo which used to loading metadata for a file regardless of its existence in Alluxio but now it doesn't load if the file is in Alluxio.